crates/sui-framework/packages/sui-system/sources/sui_system.move

```diff
@@ -611,14 +611,22 @@ module sui_system::sui_system {
         sui_system_state_inner::epoch_start_timestamp_ms(self)
     }
 
-    #[test_only]
     /// Returns the total amount staked with `validator_addr`.
     /// Aborts if `validator_addr` is not an active validator.
     public fun validator_stake_amount(wrapper: &mut SuiSystemState, validator_addr: address): u64 {
         let self = load_system_state(wrapper);
         sui_system_state_inner::validator_stake_amount(self, validator_addr)
     }
 
+    /// Returns the voting power of the active validator, values are voting power in the scale of 10000.
+    /// Aborts if `validator_addr` is not an active validator.
+    public fun validator_voting_power(wrapper: &mut SuiSystemState, validator_addr: address): u64 {
+        let validator_stake_amount = (validator_stake_amount(wrapper, validator_addr) as u128);
+        let self = load_system_state(wrapper);
+        let total_stake = (sui_system_state_inner::total_stake_amount(self) as u128);
```

Comment:
I believe what you want is the actual voting power here which is different from the stake amount: https://github.com/MystenLabs/sui/blob/315f9a6f52b44c207a99463efdbe58147e581f9e/crates/sui-framework/packages/sui-system/sources/validator.move#L142-L144

Reply:
it's different mostly because (i learnt from @emmazzz  and @lxfind):
1. voting power is capped to 10%
2. precision loss

Reply:
now using voting power from Validator struct instead